velox/dwio/dwrf/reader/SelectiveTimestampColumnReader.cpp

```diff
@@ -100,63 +115,73 @@ void SelectiveTimestampColumnReader::readHelper(RowSet rows) {
     decodeWithVisitor<velox::dwrf::RleDecoderV2<false>>(nano_.get(), visitor);
   }
   readOffset_ += numRows;
-}
 
-void SelectiveTimestampColumnReader::read(
-    vector_size_t offset,
-    RowSet rows,
-    const uint64_t* incomingNulls) {
-  prepareRead<int64_t>(offset, rows, incomingNulls);
-  VELOX_CHECK(!scanSpec_->filter());
-  bool isDense = rows.back() == rows.size() - 1;
-  if (isDense) {
-    readHelper<true>(rows);
-  } else {
-    readHelper<false>(rows);
-  }
-}
+  // Merge the seconds and nanos into 'values_'
+  auto secondsData = secondsValues_->as<int64_t>();
+  auto nanosData = values_->as<uint64_t>();
+  // If filter is present, return nulls in read to be processed.
+  auto rawNulls = nullsInReadRange_
+      ? ((returnReaderNulls_ || filter) ? nullsInReadRange_->as<uint64_t>()
```

Comment:
You have only `nullsInReadRange_` at this point

Reply:
Could you please elaborate how do these few variables interact with each other?

Reply:
`nullsInReadRange_` is the nulls get directly from this node from file.  `rawResultNulls_` is the one create that could be filtered.

Reply:
Noticed this LOC has to be kept to pass the test. Otherwise the test will fail: https://app.circleci.com/pipelines/github/facebookincubator/velox/31673/workflows/4f5ec615-5701-4544-95a4-1c88da5084d9/jobs/201442